doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,460 @@
+	Trunk []BridgeTrunkConfig `json:"trunk,omitempty"`
+	// enable setting matching vlan tags on the bridge uplink interface, default is false
+	SetUplinkVlan bool `json:"setUplinkVlan,omitempty"`
+	// MTU for VF and representor
+	MTU int `json:"mtu,omitempty"`
+	// MetaPluginsConfig configuration to be used in order to chain metaplugins
+	MetaPluginsConfig string `json:"metaPlugins,omitempty"`
+}
+
+// BridgeTrunkConfig contains configuration for OVS trunk
+type BridgeTrunkConfig struct {
+	MinID *uint `json:"minID,omitempty"`
+	MaxID *uint `json:"maxID,omitempty"`
+	ID    *uint `json:"id,omitempty"`
+}
+
+// BridgeNetworkStatus defines the observed state of BridgeNetwork
+type BridgeNetworkStatus struct {
+}
+
+// BridgeNetwork is the Schema for the ovsnetworks API
+type BridgeNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BridgeNetworkSpec   `json:"spec,omitempty"`
+	Status BridgeNetworkStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// BridgeNetworkList contains a list of BridgeNetwork
+type BridgeNetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BridgeNetwork `json:"items"`
+}
+
+```
+
+#### SriovNetworkNodePolicy CRD
+
+```golang
+// SriovNetworkNodePolicySpec defines the desired state of SriovNetworkNodePolicy
+type SriovNetworkNodePolicySpec struct {
+	// ...existing fields...
+	// contains spec for the software bridge
+	Bridge *Bridge `json:"bridge,omitempty"`
+}
+
+// contains spec for the bridge
+// only one bridge type can be set
+type Bridge struct {
+	// +kubebuilder:validation:Enum=ovs;linux
+	Type string `json:"type"`
+	// contains optional config for OVS bridge
+	Ovs *OVSConfig `json:"ovs,omitempty"`
+	// contains optional config for LinuxBr bridge
+	LinuxBr *LinuxBrConfig `json:"linux,omitempty"`
+}
+
+// OVSConfig optional configuration for OVS bridge and uplink Interface
+type OVSConfig struct {
+	Bridge    *OVSBridgeConfig    `json:"bridge,omitempty"`
+	Interface *OVSInterfaceConfig `json:"interface,omitempty"`
+}
+
+// OVSBridgeConfig contains some options from the Bridge table in OVSDB
+type OVSBridgeConfig struct {
+	DatapathType string            `json:"datapath_type,omitempty"`
+	ExternalIDs  map[string]string `json:"external_ids,omitempty"`
+	OtherConfig  map[string]string `json:"other_config,omitempty"`
+}
+
+// OVSInterfaceConfig contains some options from the Interface table of the OVSDB for PF
+type OVSInterfaceConfig struct {
+	Type        string            `json:"type,omitempty"`
+	Options     map[string]string `json:"options,omitempty"`
+	ExternalIDs map[string]string `json:"external_ids,omitempty"`
+	OtherConfig map[string]string `json:"other_config,omitempty"`
+}
+
+// LinuxBrConfig optional configuration for Linux bridge and uplink interface
+type LinuxBrConfig struct {
+	Interface map[string]string    // TODO clarify required settings
```

Comment:
can you give some examples of this field ?

Reply:
This field should hold PF's port-level bridge configuration . 
It is not clear which setting we should put here. 
VLAN configuration for the PF is a good candidate, but I'm not sure that we need it.
By default OVS doesn't require to explicitly allow VLANs on the PF interface. 
Linux bridge requires to explicitly configure VLANs on PF, but accelerated-bridge-cni already has an option to handle this.